### PR TITLE
Create convert_object_to_sorted_array.gs

### PR DESCRIPTION
**Pull request by siddhimahawar**

### CODE REVIEW

File: convert_object_to_sorted_array.gs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+* @return {Object} Sorted Array of the JSON object in descending order
+*/
+function convertObjectToSortedArray(object, processIdWithGCPId) {
+  var array = new Array();
+  for(var i in object) {
+    array.push({"key" : i, "value" : object[i], "GCPId" : processIdWithGCPId[i]});
+  }
+  
+  // get the sorted array
+  var sortedArray = sortByKey(array,"value");
```
Comment (shikhagupta1212) [inline convert_object_to_sorted_array.gs]: space after ,

File: convert_object_to_sorted_array.gs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/**
+* Convert JSON object to array sorted in descending order of their values
+* @param {Object} object will have key value pair
+* @param {Object} processIdWithGCPId will have key value pair with key as processId and value as Cloud ProjectId
+* @return {Object} Sorted Array of the JSON object in descending order
+*/
+function convertObjectToSortedArray(object, processIdWithGCPId) {
```
Comment (shikhagupta1212) [inline convert_object_to_sorted_array.gs]: Can you be specific which object it is.

File: convert_object_to_sorted_array.gs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* @return {Object} Sorted Array of the JSON object in descending order
+*/
+function convertObjectToSortedArrayForSpecificProject(object, projectId) {
+  var array = new Array();
+  for(var i in object) {
+    array.push({"key" : i, "value" : object[i], "GCPId" : projectId});
+  }
+  
+  // get the sorted array
+  var sortedArray = sortByKey(array,"value");
```
Comment (shikhagupta1212) [inline convert_object_to_sorted_array.gs]: space after ,

File: convert_object_to_sorted_array.gs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+
+/**
+* Convert JSON object to array sorted in descending order of their values
+* @param {Object} object will have key value pair
+* @param {String} projectId of the cloud project
+* @return {Object} Sorted Array of the JSON object in descending order
+*/
+function convertObjectToSortedArrayForSpecificProject(object, projectId) {
```
Comment (shikhagupta1212) [inline convert_object_to_sorted_array.gs]: I didn't get what this function is doing?

File: convert_object_to_sorted_array.gs (hunk 1 above)
Comment (siddhimahawar) [reply convert_object_to_sorted_array.gs]: Updated

File: convert_object_to_sorted_array.gs (hunk 2 above)
Comment (siddhimahawar) [reply convert_object_to_sorted_array.gs]: Updated the details of the Object

File: convert_object_to_sorted_array.gs (hunk 3 above)
Comment (siddhimahawar) [reply convert_object_to_sorted_array.gs]: Updated

File: convert_object_to_sorted_array.gs (hunk 4 above)
Comment (siddhimahawar) [reply convert_object_to_sorted_array.gs]: I removed it as it wasn't needed now

**Review by shikhagupta1212 — APPROVED** [review]

**Pull request merged**